### PR TITLE
ogr2ogr: more unique_ptr uses

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: apps/ogr2ogr_lib.cpp [hunk 1]
```diff
@@ -1662,77 +1665,69 @@ GDALVectorTranslateWrappedDataset::GDALVectorTranslateWrappedDataset(
...
     if (i < 0 || i >= static_cast<int>(m_apoLayers.size()))
         return nullptr;
-    return m_apoLayers[i];
+    return m_apoLayers[i].get();
 }
 
 OGRLayer *GDALVectorTranslateWrappedDataset::GetLayerByName(const char *pszName)
 {
-    for (size_t i = 0; i < m_apoLayers.size(); i++)
+    for (auto &poLayer : m_apoLayers)
```
Comment (dbaston) [inline apps/ogr2ogr_lib.cpp:1702]:
```suggestion
    for (const auto &poLayer : m_apoLayers)
```

File: apps/ogr2ogr_lib.cpp [hunk 2]
```diff
@@ -1662,77 +1665,69 @@ GDALVectorTranslateWrappedDataset::GDALVectorTranslateWrappedDataset(
...
     {
-        if (strcmp(m_apoLayers[i]->GetName(), pszName) == 0)
-            return m_apoLayers[i];
+        if (strcmp(poLayer->GetName(), pszName) == 0)
+            return poLayer.get();
     }
-    for (size_t i = 0; i < m_apoHiddenLayers.size(); i++)
+    for (auto &poLayer : m_apoHiddenLayers)
```
Comment (dbaston) [inline apps/ogr2ogr_lib.cpp:1707]:
```suggestion
    for (const auto &poLayer : m_apoHiddenLayers)
```

File: apps/ogr2ogr_lib.cpp [hunk 3]
```diff
@@ -1662,77 +1665,69 @@ GDALVectorTranslateWrappedDataset::GDALVectorTranslateWrappedDataset(
...
     {
-        if (strcmp(m_apoHiddenLayers[i]->GetName(), pszName) == 0)
-            return m_apoHiddenLayers[i];
+        if (strcmp(poLayer->GetName(), pszName) == 0)
+            return poLayer.get();
     }
-    for (size_t i = 0; i < m_apoLayers.size(); i++)
+    for (auto &poLayer : m_apoLayers)
```
Comment (dbaston) [inline apps/ogr2ogr_lib.cpp:1712]:
```suggestion
    for (const auto &poLayer : m_apoLayers)
```

File: apps/ogr2ogr_lib.cpp [hunk 4]
```diff
@@ -1662,77 +1665,69 @@ GDALVectorTranslateWrappedDataset::GDALVectorTranslateWrappedDataset(
...
     {
-        if (EQUAL(m_apoLayers[i]->GetName(), pszName))
-            return m_apoLayers[i];
+        if (EQUAL(poLayer->GetName(), pszName))
+            return poLayer.get();
     }
-    for (size_t i = 0; i < m_apoHiddenLayers.size(); i++)
+    for (auto &poLayer : m_apoHiddenLayers)
```
Comment (dbaston) [inline apps/ogr2ogr_lib.cpp:1717]:
```suggestion
    for (const auto &poLayer : m_apoHiddenLayers)
```

File: apps/ogr2ogr_lib.cpp [hunk 5]
```diff
@@ -1662,77 +1665,69 @@ GDALVectorTranslateWrappedDataset::GDALVectorTranslateWrappedDataset(
...
     OGRLayer *poLayer = m_poBase->GetLayerByName(pszName);
     if (poLayer == nullptr)
         return nullptr;
-    poLayer = GDALVectorTranslateWrappedLayer::New(poLayer, false,
-                                                   m_poOutputSRS, m_bTransform);
-    if (poLayer == nullptr)
+
+    auto poNewLayer =
+        std::unique_ptr<OGRLayer>(GDALVectorTranslateWrappedLayer::New(
+            poLayer, /* bOwnBaseLayer = */ false, m_poOutputSRS, m_bTransform));
```
Comment (dbaston) [inline apps/ogr2ogr_lib.cpp:1729]:
```suggestion
    auto poNewLayer = GDALVectorTranslateWrappedLayer::New(
            poLayer, /* bOwnBaseLayer = */ false, m_poOutputSRS, m_bTransform);
```

**Review by dbaston — APPROVED** [review]

File: apps/ogr2ogr_lib.cpp (hunk 2 above)
Comment (rouault) [reply apps/ogr2ogr_lib.cpp]: done

File: apps/ogr2ogr_lib.cpp (hunk 3 above)
Comment (rouault) [reply apps/ogr2ogr_lib.cpp]: done

File: apps/ogr2ogr_lib.cpp (hunk 4 above)
Comment (rouault) [reply apps/ogr2ogr_lib.cpp]: done

File: apps/ogr2ogr_lib.cpp (hunk 5 above)
Comment (rouault) [reply apps/ogr2ogr_lib.cpp]: done

**Pull request merged**